test/e2e/specs/site-editor/site-editor-inserter.spec.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site Editor Inserter', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.activateTheme( 'emptytheme' ),
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'inserter toggle button should toggle global inserter', async ( {
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Toggle block inserter"i]' );
+		await expect(
+			page.locator( 'role=region[name="Block Library"i]' )
+		).toBeVisible();
```

Comment:
We still need the other steps: clicking on the button again and expecting that it's hidden.

Reply:
I added both the steps and added role locators for all the assertions. Thanks for guiding me and helping me out